mmv1/templates/terraform/examples/network_security_tls_inspection_policy_custom.tf.erb

```diff
@@ -0,0 +1,136 @@
+data "google_project" "project" {
+  provider = google-beta
+}
+
+resource "google_privateca_ca_pool" "default" {
+  provider = google-beta
+  name      = "<%= ctx[:vars]['privateca_ca_pool_name'] %>"
+  location  = "us-central1"
+  tier     = "DEVOPS"
+  publishing_options {
+    publish_ca_cert = false
+    publish_crl = false
+  }
+  issuance_policy {
+    maximum_lifetime = "1209600s"
+    baseline_values {
+      ca_options {
+        is_ca = false
+      }
+      key_usage {
+        base_key_usage {}
+        extended_key_usage {
+          server_auth = true
+        }
+      }
+    }
+  }
+}
+
+
+resource "google_privateca_certificate_authority" "default" {
+  provider = google-beta
+  pool = google_privateca_ca_pool.default.name
+  certificate_authority_id = "<%= ctx[:vars]['privateca_certificate_authority_id'] %>"
+  location = "us-central1"
+  lifetime = "86400s"
+  type = "SELF_SIGNED"
+  deletion_protection = false
+  skip_grace_period = true
+  ignore_active_certificates_on_deletion = true
+  config {
+    subject_config {
+      subject {
+        organization = "Test LLC"
+        common_name = "my-ca"
+      }
+    }
+    x509_config {
+      ca_options {
+        is_ca = true
+      }
+      key_usage {
+        base_key_usage {
+          cert_sign = true
+          crl_sign = true
+        }
+        extended_key_usage {
+          server_auth = false
+        }
+      }
+    }
+  }
+  key_spec {
+    algorithm = "RSA_PKCS1_4096_SHA256"
+  }
+}
+
+resource "google_project_service_identity" "ns_sa" {
+  provider = google-beta
+
+  service = "networksecurity.googleapis.com"
+}
+
+resource "google_privateca_ca_pool_iam_member" "tls_inspection_permission" {
+  provider = google-beta
+
+  ca_pool = google_privateca_ca_pool.default.id
```

Comment:
Try to make this `google_privateca_certificate_authority.default.pool`.
Not sure if the name will work. Alternatively, you can try to pass the id instead of the pool name in `google_privateca_certificate_authority`.

If you make this work, all explicit dependencies below (depends_on) will go away.

Reply:
google_privateca_certificate_authority does not work with id